src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -110,134 +113,68 @@ public ArmSubsystem() {
         m_shoulder.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, -15);
         m_elbow.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, 60);
         m_elbow.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, -15);
+    }
 
+    public void setState(State desiredState) {
+        this.m_desiredState = desiredState;
     }
 
-    public void setState(State state) {
-        // Don't do anything already in correct state
-        if (state == m_state_actual)
-            return;
-
-        switch (m_state_actual) {
-            case Pickup:
-                if (state == State.Rest)
-                    m_state_desired = state;
-                break;
-            case Rest:
-                m_state_desired = state;
-                break;
-            case PlacingA:
-            case PlacingB:
-            case PlacingC:
-            case PlacingD:
-            case PlacingE:
-                m_state_desired = state;
-                break;
-            case Transit:
-                break;
-        }
+    public boolean atState() {
+        boolean shoulder_at_state = goalCalculator(m_shoulder.getEncoder().getPosition(), ArmConstants.SHOULDER_GEAR_RATIO(m_desiredState.shoulder));
+        boolean elbow_at_state = goalCalculator(m_elbow.getEncoder().getPosition(), ArmConstants.ELBOW_GEAR_RATIO(m_desiredState.elbow));
+        return shoulder_at_state && elbow_at_state;
+    }
+
+    private boolean goalCalculator(double current, double desired) {
+        if (current == desired) return true;
+        return Math.abs(current - desired) <= 10;
     }
 
     public ArmSubsystem.State deserializeStateFromString(String stateName) {
         System.out.println("Deserializing state for " + stateName);
 
         switch (stateName) {
-            case "Rest": return State.Rest;
-            case "Pickup": return State.Pickup;
-            case "PlacingA": return State.PlacingA;
-            case "PlacingB": return State.PlacingB;
-            case "PlacingC": return State.PlacingC;
-            case "PlacingD": return State.PlacingD;
-            case "PlacingE": return State.PlacingE;
+            case "Rest":
+                return State.Rest;
+            case "Pickup":
+                return State.Pickup;
+            case "PlacingA":
+                return State.PlacingA;
+            case "PlacingB":
+                return State.PlacingB;
+            case "PlacingC":
+                return State.PlacingC;
+            case "PlacingD":
+                return State.PlacingD;
+            case "PlacingE":
+                return State.PlacingE;
         }
 
         return ArmSubsystem.State.Rest;
     }
 
     public void periodic() {
-        matchActualState();
-        POS position;
-        PID shoulder_pid;
-        PID elbow_pid;
-
-        // If the state is not the same as the desired state, then run the periodic for that state.
-        if (m_state_actual != m_state_desired) {

```

Comment:
We need two different PID's one for steady state and one for large movements. They are tuned differently and it is not feasible to tune a single PID for both cases.

Reply:
Ok, then why not use the PID classes instead?

Reply:
Or you could switch between slots.

Reply:
That is what was happening before.

Reply:
What's stopping us from doing so now?

Reply:
Nothing but your code needs that to be re implemented. Hence the review.